src/lib/__tests__/InstantSearch-test.tsx

```diff
@@ -540,6 +540,31 @@ describe('addWidget(s)', () => {
 
     expect(search.addWidgets([createWidget()])).toBe(search);
   });
+
+  it('does not trigger a search with initial results', async () => {
+    const searchClient = createSearchClient();
+    const search = new InstantSearch({
+      indexName: 'indexName',
+      searchClient,
+    });
+    // An empty object is a valid value for initial results
+    search._initialResults = {};
+
+    search.start();
+    search.addWidgets([createWidget()]);
```

Comment:
I assume this is to test adding widgets in the same tick after the instance is started. Is it worth also testing adding widgets before, then starting?